### PR TITLE
Fixes #82: Allow ordering records list by column.

### LINKED ISSUE #82: (minor) Validate collection id/name
capture d ecran de 2016-05-24 11-50-16

Idea: use the current field to populate a `label` attribute, and slugify to assign the `id` attribute to be used.

### PR DESCRIPTION
**Pull request by n1k0**
Refs #82, this is WiP and ~~doesn't work just yet~~ does work now (I think).

Note: this also fixes an unreported bug where we're loading the list of records **twice** each time we're accessing the related route.

#### Todo

- [x] Basic sorting implementation
- [x] Tests
- [x] Hide sort links for non-schema or nested fields as they're not supported
- [x] Store default sort value in collection data?

### CODE REVIEW

File: scripts/components/CollectionList.js [hunk 1]
```diff
@@ -58,6 +58,32 @@ class Row extends Component {
   }
 }
 
+function SortLink(props) {
+  const {dir, column, updateSort} = props;
+  return (
+    <a href="#" className="sort-link" onClick={(event) => {
+      event.preventDefault();
+      updateSort(dir === "up" ? `-${column}` : column);
```
Comment (Natim) [inline scripts/components/CollectionList.js]: I don't understand how `up` means `-$column`

Comment (n1k0) [inline scripts/components/CollectionList.js]: This is a toggler so we need the action opposite to current state.

**Comment by n1k0** [comment]
I'm now taking early feedback. Deployed the patch to gh-pages. Yeah, we badly need a web designer :cry: 

/cc @glasserc @leplatrem @Natim

**Comment by Natim** [comment]
Let me have a look

**Comment by Natim** [comment]
I think went we activate the ordering it should stay ASC as the arrow shows it.

**Comment by leplatrem** [comment]
Nice! GG!

In term of UX, I have one remark: it is a bit troubling that the whole array disappears behind the loader.

If feasible and not too costly, keeping the headers row would help keeping the mental context of the click :)

**Comment by n1k0** [comment]
> In term of UX, I have one remark: it is a bit troubling that the whole array disappears behind the loader.

I agree though that will be addressed in a followup PR as there's a bunch spinner mess to clean and this would probably bloat up this PR :)

**Comment by n1k0** [comment]
Patch is complete I think. May I now ask for a review of the code? :) /cc @leplatrem @Natim @glasserc

**Comment by n1k0** [comment]
I've been self-reviewing this patch a lot and am pretty sure it's ready for landing, so I'm gonna land this so I can move on.

**Pull request merged**